cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextTranslator.scala

```diff
@@ -76,9 +78,15 @@ trait TextAsOnlyEntity extends HasTextInput with HasCognitiveServiceInput {
 
   override protected def prepareEntity: Row => Option[AbstractHttpEntity] = {
     r =>
-      Some(new StringEntity(
-        getValueOpt(r, text)
-          .map(x => x.map(y => Map("Text" -> y))).toJson.compactPrint, ContentType.APPLICATION_JSON))
+      val textVal = getValueOpt(r, text)
+      if (textVal.nonEmpty) {
+        val content = textVal.get.getClass.getName match {
```

Comment:
Chris' second issue was caused by having nulls within a batch. Perhaps we should add that case as a test and ensure this new func can handle. I think in my TA batching logic it had to get pretty hairy to handle that unfortunately. Perhaps we can use similar logic. Also we might want to consider a better solution that just batching elements into single arrays. But we can push that to a later PR as that is current behavior in TA

Reply:
I think the error comes from when there are nulls of text which would ordinarily be skipped but they are in a batch so the mess up the whole batch